chain/network/src/peer/codec.rs

```diff
@@ -97,6 +114,7 @@ impl Decoder for Codec {
             if buf.is_empty() && buf.capacity() > 0 {
                 *buf = BytesMut::new();
             }
+            self.buf_size_metric.set(buf.len() as i64);
```

Comment:
why do you set it twice? Does consuming the buffer content may trigger buffer size change? Anyway calling it when exiting the function should suffice, right?

Reply:
Added comment in code.